torch/csrc/distributed/c10d/logging.cpp

```diff
@@ -34,4 +34,20 @@ bool isLogLevelEnabled(LogLevel level) noexcept {
   return false;
 }
 
+void lockWithLogging(
+    std::unique_lock<std::timed_mutex>& lock,
+    std::chrono::milliseconds log_interval,
+    c10::string_view desc,
+    c10::string_view file,
+    int line) {
+  while (!lock.try_lock_for(log_interval)) {
```

Comment:
I am wondering if we can just do `try_lock_for` just once and then try to capture the lock like what we do in the past?

Reply:
yeah, that's fair -- just depends if it's too much log spam